src/buildmanager.cpp

```diff
@@ -2115,9 +2129,20 @@ QString BuildManager::findFile(const QString &defaultName, const QStringList &se
 			QString absPath = base.absolutePath() + "/";
 			QString baseName = "/" + base.fileName();
 			QFileInfo fi(absPath + p + baseName);
-			if (fi.exists()) return fi.absoluteFilePath();
-		}
+            if (fi.exists()) {
+                if (mostRecent) {
+                    if (mr == nullptr || mr->lastModified() < fi.lastModified()) {
+                        if (mr != nullptr)
+                            delete mr;
+                        mr = new QFileInfo(fi);
+                    }
+                } else
+                    return fi.absoluteFilePath();
+            }
+        }
 	}
+	if (mostRecent && mr != nullptr)
+        return mr->absoluteFilePath();

```

Comment:
I guess that you have a resource leak here ...
and please use indentation, better braces, to connect which return is called when

Reply:
Oh yes ; I ordinarily program in Python or OCaml, so manual memory management is not something I am used to !

It should be better with my most recent commit.

Reply:
Indentation was inconsistent (tabs/spaces). I switched back to tab-indentation as was the case originally in that method (however other methods in the same file are space-indented…).

Reply:
C++ has  automatic memory management, too. Pointer variables just needs to be declared with `unique_ptr<QFileInfo> mr` rather than `QFileInfo* mr ` and assigned with the constructor/reset rather than `=`. Or here an empty QFileInfo  could have been used rather than a pointer and checked if it exists rather than if it is null